src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -114,7 +115,8 @@
    :version s/Str
    :environment (s/maybe s/Str)
    :transaction_uuid (s/maybe s/Str)
-   :producer_timestamp (s/maybe pls/Timestamp)
+   :producer_timestamp (s/pred #(not (nil? (to-timestamp %))) 'coercable-to-timestamp)
```

Comment:
Updated to be `pls/Timestamp`, and checked that it gives a good message on invalid input. 

```
Value does not match schema: {:producer_timestamp (not (datetime? a-java.lang.String))}
```